service/history/timerQueueProcessor.go

```diff
@@ -654,17 +654,32 @@ Update_History_Loop:
 				}
 			} else {
 				// See if we have next timer in list to be created.
-				if !td.TaskCreated || td.ActivityID == scheduleID {
+				isHeartBeatTask := timerTask.TimeoutType == int(workflow.TimeoutType_HEARTBEAT)
+
+				// Create next timer task if we don't have one (or)
+				// if current one is HB task and we need to create next HB task for the same.
+				if !td.TaskCreated || (isHeartBeatTask && td.ActivityID == scheduleID) {
```

Comment:
don't want to do that as we spoke offline. that creates that timer queue processor dependency on the frequency of user updates.  let say if i have ~3min HB timer created, if the user keeps reporting earlier, we might clear the flag and timer q processor might think for the minimum timer doesn't exist and it keeps creating the timer for this. this would happen if a new minimum keeps coming in.